src/applications/hca/utils/helpers/listloop-pattern.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Helper that takes query params and sets labels and return paths for
+ * the multiresponse (list/loop) information pages
+ * @param {Object} props - the original dataset, key name, localData object,
+ * search index, view field object and ref item for the src array
+ * @returns {Object} - the dataset to return
+ */
+export function getDataToSet(props) {
+  const { slices, dataKey, localData, listRef, viewFields } = props;
+  return localData === null
```

Comment:
Would something like this be a little more readable here than the ternary with the big object it returns? 

```
  const dataSet = localData
    ? [...slices.beforeIndex, localData, ...slices.afterIndex]
    : listRef;

  return {
    [dataKey]: dataSet,
    [viewFields.report]: null,
    [viewFields.skip]: true,
  };
```
I'm not sure if we only care if localData is null or just something that is falsy (like undefined, 0, or false), but this change looks like it passes all of the specs and DRYs up some of the return. If we only want to validate on null we could invert the dataSet like this:
```
const dataSet =
  localData === null
    ? listRef
    : [...slices.beforeIndex, localData, ...slices.afterIndex];
```